feat : 전체 상품 목록 조회 기능 구현

asd42270 opened the pull request:
AccommodationDto : 숙박 정보 엔티티 클래스(참조 관계 미구현)
AccommodationType : 숙소의 Type을 나타내는 enum 클래스
SpringDataJpaAccommodationRepository : AccommodationDto 레포지토리 인터페이스
SpringDataJpaAccommodationRepositoryTest : 레포지토리 테스트 클래스- 테스트 완료

## #️⃣연관된 이슈

> ex) #1 
> 레포지토리 단위 테스트 완료
## 📝작업 내용

> 이번 PR에서 작업한 내용을 간략히 설명해주세요(이미지 첨부 가능)
> - [ ]  숙박 DTO 구현
> - [ ] 숙박 Repository 구현 및 테스트 완료
> - [ ] Service 클래스 구현 완료 
> - [ ] Service 클래스 테스트 진행 중

## 💬추가 사항
> 리뷰에 대한 상세 내용이나 Reviewer에 대한 요청 사항 등 필요한 부분을 자유롭게 작성해주세요
> - [ ] 서비스 클래스 구현하면서 BaseResponse 관련해서 궁금한 점이 생겨서 내일 한 번 여쭤보려고 합니다!!

asd42270 committed:
feat : 전체 상품 목록 조회 기능 구현
AccommodationDto : 숙박 정보 엔티티 클래스(참조 관계 미구현)
AccommodationType : 숙소의 Type을 나타내는 enum 클래스
SpringDataJpaAccommodationRepository : AccommodationDto 레포지토리 인터페이스
SpringDataJpaAccommodationRepositoryTest : 레포지토리 테스트 클래스- 테스트 완료

Woohahalife commented:
AccommodationDto를 entity로 설정하셨던데 entity 객체와 dto 객체는 구분해서 쓰셔야해용!
entity 클래스 단일로 사용을 해버리면 이후 연관관계 설정할 때 무한 loop가 발생해버리구 dto를 사용하지 않는다면 요청 & 응답값을 모델링할 때 적절한 데이터만 선별할 수가 없게됩니당.

asd42270 committed:
feat : 전체 상품 목록 조회 기능 구현
AccommodationDto -> Accommodation(클래스 이름 변경) : 숙박 정보 엔티티 클래스(Room 참조 관계 구현)
Room : Accommodation과 참조 관계(하나의 숙소는 여러 개의 방을 가질 수 있다.)
SpringDataJpaAccommodationRepository : Room 관계 추가
SpringDataJpaAccommodationRepositoryTest : Room 관계 추가 후 테스트 완료          SpringDataJpaRoomRepository : Room 엔티티 레포지토리
